src/blueprint/into_schema.rs

```diff
@@ -102,11 +160,23 @@ fn to_type(def: &Definition) -> dynamic::Type {
                       const_value
                     }
                   }
-                  _ => expr.eval(&ctx, &Concurrent::Sequential).await?,
+                  _ => {
+                    if let Ok(const_value) = read_entity_cache(&ctx, &of_type).await {
+                      read_from_entity_cache = true;
+                      log::info!("Reading to cache.");
+                      const_value
+                    } else {
+                      expr.eval(&ctx, &Concurrent::Sequential).await?
+                    }
+                  }
```

Comment:
I thought in `src/config/from_document.rs` we copy cache settings from object type down to fields, so settings are spread to every field. It doesn't work that way how I think? If I'm right then there couldn't be a situation when we have an object level cache and not field level

Reply:
The way the implementation works currently is that the cache directives at field will work only for the fields which have resolvers. If we want a type to be cached irrespective of the query fetching it, then we will need to keep it on the object level and it is used globally by any query or field which will output that type

Reply:
now I get it.

but anyway if we still copy cache settings to field level is `_ =>` condition will apply? It will probably be still something there

Reply:
Good point. I think now since we have an additional way of caching at entity level, it is better to not make the fields inherit the cache setting provided at the type. Directives provided at field will use the old way of caching and the ones at the type will use the new implementation. Does this sound like the right way of doing it? @tusharmath @amitksingh1490

Reply:
Let's think again about the bug and the fix because current code looks strange for me.

For single field you just now have 2 caches with unclear relation. Consider the case when caches are empty:
1. ctx.req_ctx.cache_get for field level cache returns None
2. read_entity_cache for type level cache returns None
3. then we execute else branch that writes field level cache
4. since read_from_entity_cache is false we write also to type level cache

At the end we have same value in two caches and we'll use only first one anyway.

Consider now when you want to remove copying `@cache` directive from type to field. In that case for such fields we won't create field level cache and will use type level cache. Seems ok, but it's actually the same as before just with different cache entity.

I see the example in the parent issue like this
```
type Query {
  posts: [Post] @http(path: "/posts")
}

type Post {
  id: Int! @cache(maxAge: 3000)
  userId: Int! @cache(maxAge: 4000)
  title: String!
  body: String!
  user: User @http(path: "/users/{{value.userId}}")
}
```

but it's not clear for me what are we going to cache here? constant values that anyway will be resolved by actual http parent request? what are the benefits of caching values that will be resolved by something else?

Reply:
The limitation of the cache directive at field level, right now, is that we cannot cache the values which do not have some resolver, because the code responsible for caching the output of the field doesn't get executed for them. In the above example, only if cache is applied for the `posts` query or the `user` field inside `Post` type will it be able to cache the output. This can be fixed if while storing the cache we split the values into separate fields and store them separately. This would require us to store the cache settings globally just like I'm doing for this global cache.
Also, the reason we are having to store same value at two separate caches is that, the first cache will only work for that particular query or field but the second one will work for any query or field which output's that type. And also the second cache will not be populated if cache isn't applied to the output type of that query. The way I am thinking about this is that, if the user wants to cache a type, irrespective of the query which outputs it, then they will only provide cache setting for that type, and if they only want to cache output of a query, irrespective of the type which it outputs then they will provide it at that query

Reply:
I've actually misread the code in terms what is type_name is used to check the cache. It quite confusing that while parsing config we copy cache settings from type to all the fields of that type. And here we check the type of the field itself, not the wrapping type. Am I right?

Also, I think example from the issue is misleading since the fields without resolver are actually resolved with another branch above i.e.

```
match &field.resolver {
  None => {
    let ctx = EvaluationContext::new(req_ctx, &ctx);
    FieldFuture::from_value(ctx.path_value(&[field_name]).map(|a| a.to_owned()))
  }
```

and that related to values that is already resolved by parent value with its resolver and there is no benefits of caching those.